### PR TITLE
TECS throttle gains default reduction and transition values from previous PX4 version

### PR DESCRIPTION
**Pull request by sfuhrer**
### Solved Problem
Fixes https://github.com/PX4/PX4-Autopilot/issues/22540.
TL;DR: PX4 v1.12 and v1.13 contained a bug that resulted in FW_T_I_GAIN_THR and FW_T_THR_DAMP having a much lower effect. This bug is fixed now, resulting in these params to have much stronger effects on the controller, which can lead to controller instability. 

### Solution
- reduce the defaults
- introduce mechanism to warn user if gains are high    
  - The params `FW_T_I_GAIN_THR` and `FW_T_THR_DAMP` have a much stronger effect  in PX4 v1.14 then before, and therefore we warn the user if the current values are above normally reasonable values, unless they have acknowledged  knowing about the change by setting the new param `FW_T_CHNG_ACK`.

### Changelog Entry
For release notes:
```
Feature: FW TECS: Warn if throttle gains (that have changed effect with this release) are high
Documentation: To do.
```

### Alternatives
- Deprecate the old params `FW_T_I_GAIN_THR` and `FW_T_THR_DAMP` and in their place make new ones (on v1.14 and main). Then, on updating, previous tuning gets reset to default, which is much better than keeping it. Alternatively we could do some sort of param translation (eg with a factor of 20). --> this is though not optimal for users already flying with v1.14 and thus have re-tuned the gains, for them they would also get adjusted. 
- limit the params to sane values (hard coded, not just in param meta data)

### CODE REVIEW

**Comment by tstastny** [comment]
I'm not the biggest fan of the extra acknowledge param. Though I see how this is tricky to handle otherwise. I suppose in 1.15 we would deprecate the ack param? I would probably err on the side of just documenting it in the patch release changelog, as generally a patch release implies there is an urgent bug fix that folks need to ingest. Then also in the minor version change log where we should anyway be talking about param translations and major diffs. This would then cover both existing 1.14 users and those upgrading from older versions. Of course relies on people reading docs and/or change logs..

@bkueng @dagar wdyt about this? Is there any other way you can think to handle it?

**Comment by sfuhrer** [comment]
>We probably shouldn't have kept the same parameter naming unless there was enough testing in place before and after to be sure that's valid.

Yes clearly. 

>Ideally we'd be able catch the actual upgrade (eg 1.13->1.14) and flag anything relevant like this for review.

Do we have a way to catch the upgrade? And then do an automatic param translation? One hacky way I was thinking about was checking in the param translation about the existence of a param that is only existing in v1.13, and use that to catch to upgrade. 

@dagar would you otherwise also just go with documentation in the release notes?

**Comment by sfuhrer** [comment]
As discussed in the maintainer call yesterday, I changed the PR: removed the warning mechanism and instead renamed the two affected parameters. Further we discussed that we will NOT port anything to v1.14 to not introduce a change between v1.14.0 and v1.14.1. Will instead mention it in the 1.14 release notes.

**Comment by sfuhrer** [comment]
As discussed in the maintainer call yesterday, I changed the PR: removed the warning mechanism and instead renamed the two affected parameters. Further we discussed that we will NOT port anything to v1.14 to not introduce a change between v1.14.0 and v1.14.1. Will instead mention it in the 1.14 release notes.
@tstastny @dagar

**Review by tstastny — DISMISSED** [review]

**Review by tstastny — APPROVED** [review]

**Comment by ryanjAA** [comment]
When is a good idea to test this? Meaning I’m at the default values on 1.14 at 0.05 but the new values reduce this down further (0.02). Is this something I need to go and fly at 0.02 or just wait until the next round of testing or changes to look at? 

And why are we decreasing from 0.05 to 0.02? I understand that the integrator has more impact now but was there testing to suggest we should decrease this?

**Comment by sfuhrer** [comment]
> When is a good idea to test this? Meaning I’m at the default values on 1.14 at 0.05 but the new values reduce this down further (0.02). Is this something I need to go and fly at 0.02 or just wait until the next round of testing or changes to look at?
> 
> And why are we decreasing from 0.05 to 0.02? I understand that the integrator has more impact now but was there testing to suggest we should decrease this?

Yes our testing implied that further reducing it results in a more appropriate throttle dynamics. Otherwise you end up with seeing every wind gust in the integrator value, which is not what it is meant for (it's mainly meant for cancelling out slow disturbances in the motor model like depleting battery). 

It's a good time to start using the new params with the new defaults now. Please let us know if you observe regressions with it.
